store/sqlstore/channel_store.go

```diff
@@ -998,6 +998,83 @@ func (s SqlChannelStore) GetChannels(teamId string, userId string, includeDelete
 	return channels, nil
 }
 
+func (s SqlChannelStore) GetChannelsByUser(userId string, includeDeleted bool, lastDeleteAt, pageSize int, fromChannelID string) (model.ChannelList, error) {
+	query := s.getQueryBuilder().
+		Select("Channels.*").
+		From("Channels, ChannelMembers").
+		Where(
+			sq.And{
+				sq.Expr("Id = ChannelId"),
+				sq.Eq{"UserId": userId},
+			},
+		).
+		OrderBy("Id ASC")
+
+	if fromChannelID != "" {
+		query = query.Where(sq.Gt{"Id": fromChannelID})
+	}
+
+	if pageSize != -1 {
+		query = query.Limit(uint64(pageSize))
+	}
+
+	if includeDeleted {
+		if lastDeleteAt != 0 {
+			// We filter by non-archived, and archived >= a timestamp.
+			query = query.Where(sq.Or{
+				sq.Eq{"DeleteAt": 0},
+				sq.GtOrEq{"DeleteAt": lastDeleteAt},
+			})
+		}
+		// If lastDeleteAt is not set, we include everything. That means no filter is needed.
+	} else {
+		// Don't include archived channels.
+		query = query.Where(sq.Eq{"DeleteAt": 0})
+	}
+
+	sql, args, err := query.ToSql()
+	if err != nil {
+		return nil, errors.Wrapf(err, "getchannels_tosql")
+	}
+
+	var channels model.ChannelList
+	_, err = s.GetReplica().Select(&channels, sql, args...)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get channels with UserId=%s", userId)
+	}
+
+	if len(channels) == 0 {
+		return nil, store.NewErrNotFound("Channel", "userId="+userId)
+	}
+
+	return channels, nil
+}
+
+func (s SqlChannelStore) GetAllChannelMembersById(id string) ([]string, error) {
+	perPage := 500
+	page := 0
+
+	var res []string
+	for {
+		channelMembers, err := s.GetMembers(id, page*perPage, perPage)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, member := range channelMembers {
+			res = append(res, member.UserId)
+		}
+
+		if len(channelMembers) < perPage {
+			break
+		}
+
+		page++
+	}
```

Comment:
Is there an actual benefit/need to page through here? Can't we simply run a query to return them all at once?

Reply:
Good point. What I was thinking is that, purely from a theoretical perspective, the total amount of required DB memory would be slightly larger than just the number of rows, since that query has to join 5 tables. OTOH, the server just has to handle purely the results. And moreover, since this is only called from a job, it's okay to spread the load on the DB slightly.

I don't have 100% experimental proof though. Just using the force. Okay to fetch full results if you think it'll be better.

Reply:
I think your theory would work but only if the limiting of data was done in the JOINs. Looking at the query we are just limiting the final set of rows so from a DB perspective there wouldn't be any performance benefit of re-running the query more than once as it's the same work again no matter the page size.

Reply:
Of course, you are right.